File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    for (let i = 0; i < nums.length; i++){
        if (nums[i] == 0){
            let j = i + 1;
            while (j < nums.length){
                if (nums[j] !== 0){
                    let temp = nums[j];
                    nums[j] = 0;
                    nums[i] = temp;
                    break;
                }
                j++;
            }
        }
    }
    return nums;
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        insert_pos = 0
        n = len(nums)

        for i in range(n):
            if nums[i] !=0:
                nums[insert_pos] = nums[i]
                insert_pos += 1
        

        for i in range(insert_pos, n):
            nums[i] = 0
        